mlir/lib/Conversion/FuncToEmitC/FuncToEmitC.cpp

```diff
@@ -53,13 +60,34 @@ class FuncOpConversion final : public OpConversionPattern<func::FuncOp> {
   matchAndRewrite(func::FuncOp funcOp, OpAdaptor adaptor,
                   ConversionPatternRewriter &rewriter) const override {
 
-    if (funcOp.getFunctionType().getNumResults() > 1)
+    FunctionType type = dyn_cast<FunctionType>(funcOp.getFunctionType());
```

Comment:
Is the cast needed? I see getFunctionType already returning a `FunctionType`

Reply:
I think the generic function converter pass needed to do it because other `FuncOp`s were supported than just the one from `func` dialect. Here we don't need that anymore indeed.